lib/netext/transport.go

```diff
@@ -0,0 +1,102 @@
+package netext
+
+import (
+	"net"
+	"net/http"
+	"strconv"
+
+	"github.com/loadimpact/k6/lib"
+	"github.com/loadimpact/k6/stats"
+	"github.com/pkg/errors"
+)
+
+type Transport struct {
+	http.RoundTripper
+	options   lib.Options
+	tags      map[string]string
+	trail     *Trail
+	tlsInfo   TLSInfo
+	samplesCh chan<- stats.SampleContainer
+}
+
+func NewTransport(transport http.RoundTripper, samplesCh chan<- stats.SampleContainer, options lib.Options, tags map[string]string) *Transport {
+	return &Transport{
+		RoundTripper: transport,
+		tags:         tags,
+		options:      options,
+		samplesCh:    samplesCh,
+	}
+}
+
+func (t *Transport) SetOptions(options lib.Options) {
+	t.options = options
+}
+
+func (t *Transport) GetTrail() *Trail {
+	return t.trail
+}
+
+func (t *Transport) TLSInfo() TLSInfo {
+	return t.tlsInfo
+}
+
+func (t *Transport) RoundTrip(req *http.Request) (res *http.Response, err error) {
+	if t.RoundTripper == nil {
+		return nil, errors.New("no roundtrip defined")
+	}
+	tags := t.tags
```

Comment:
I think this would cause k6 to mess up the tags if there are redirects, since they won't be copies and every request would modify the original map

Reply:
Illustration of the issue: https://play.golang.org/p/to1AYVVJ_vN
The only way I know that we can prevent that mangling is to copy the map values into a new map...